File: crates/vybe_host/src/namespaces/math.rs
use super::*;

pub fn register(vm: &mut VM) {
    // Math (direct)
    let math = ensure_namespace(vm, &["Math"]);
    for name in &[
        "floor", "ceil", "round", "abs", "sqrt", "pow", "min", "max",
        "sin", "cos", "tan", "log", "sign", "trunc", "log2", "log10",
        "cbrt", "hypot", "atan2", "asin", "acos", "atan", "exp", "clz32",
    ] {
        set_prop(&math, name, host_fn_ref(vm, "ecma:math", name));
    }
    set_prop(&math, "pi", Value::F64(std::f64::consts::PI));
    set_prop(&math, "e", Value::F64(std::f64::consts::E));
}

File: examples/js/lib/math.js
export function clamp(value, min, max) {
    if (value < min) return min;
    if (value > max) return max;
    return value;
}

export function lerp(a, b, t) {
    return a + (b - a) * t;
}

export function distance(x1, y1, x2, y2) {
    let dx = x2 - x1;
    let dy = y2 - y1;
    return Math.sqrt(dx * dx + dy * dy);
}
